### ISSUE TITLE
Eltharion can imprison an opponent even when he loses

### ISSUE THREAD

**Issue by LordOmlette** [chadvandy/cbfm_wh3#484]
**Describe the bug**

It's possible when playing as Yvresse to imprison an opponent even if you lost if you autoresolve the battle.

**Expected behavior**

Imprisoning an opponent via autoresolve should *only* happen if Yvresse was attacking *and* won.

**To Reproduce**

> Playing Eltharion: is anyone else getting behaviour where they suicide a lord into a stack and autoresolve it, and it captures their damn general? I just kidnapped N'Kari trying to suicide a lord and I don't have the success on autoresolve mod.
> -Lycia Pintella

1. Play as Yvresse.
2. Remove all units from the starting High Elf Princess' units.
3. Attack the nearby greenskin force.
4. There's a 50% chance you will imprison them anyway despite losing.

This repro is problematic but the autoresolve capture chance is hardcoded @ 50%.

**Additional context**

In wh2_dlc15_eltharion_lair.lua, there's an event listener named "lair_BattleCompleted".  The event filter is for whether or not Yvresse is human.

```
return cm:get_faction(yvresse_faction_key):is_human()
```

It should be enough to also check if the attacker won:

```
return cm:get_faction(yvresse_faction_key):is_human() and cm:model():pending_battle():attacker_won()
```

**Comment by LordOmlette** [comment]
![image](https://user-images.githubusercontent.com/112673788/195394713-430181f1-43cb-46c9-9c83-dec8cb6a0d39.png)
[Yvresse_Quick Save.13585111484.zip](https://github.com/chadvandy/cbfm_wh3/files/9766825/Yvresse_Quick.Save.13585111484.zip)

**Comment by LordOmlette** [comment]
This is not fixed in patch 2.4.0.

**Comment by mfvreeland** [comment]
Is this fixed in 4.1?

**Comment by LordOmlette** [comment]
This is not fixed in patch 4.1.0.

**Comment by mfvreeland** [comment]
Thanks for confirming. I will take a look at this one.
